### DISCUSSION TITLE
Cannot make proxy without auth work, while with auth working OK.

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3631] [Q&A]
Hi! I create browser in a following manner:

`Browser = new ChromiumWebBrowser();
           
            (Browser as ChromiumWebBrowser).BrowserSettings = browserSetting;
            //Using RequestContextSettings allows having isolated cache/cookies etc for different ChromiumWebBrowser instances
            RequestContextSettings requestContextSettings = new RequestContextSettings();
            requestContextSettings.PersistSessionCookies = true;
            requestContextSettings.PersistUserPreferences = true;
          
            string cachePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments),
                UniCefBrowserWrapperFactory.CefSharpCacheLocalPath,
                SingleBrowserInfo.Proxy.Ip);
            if (!Directory.Exists(cachePath)) Directory.CreateDirectory(cachePath);
            requestContextSettings.CachePath = cachePath;
            (Browser as ChromiumWebBrowser).RequestContext = new RequestContext(requestContextSettings);
            (Browser as ChromiumWebBrowser).RequestHandler =
                new CustomRequestHandler(SingleBrowserInfo.Proxy.Login,
                    SingleBrowserInfo.Proxy.Password,
                    SingleBrowserInfo.UserAgent);
            if (!String.IsNullOrEmpty(SingleBrowserInfo.Proxy.Ip))
                (Browser as ChromiumWebBrowser).IsBrowserInitializedChanged += (s, e) =>
                {
                    var br = (ChromiumWebBrowser)s;
                    var v = new Dictionary<string, object>
                    {
                        ["mode"] = "fixed_servers",
                        ["server"] = string.Format("{0}://{1}:{2}",
                        SingleBrowserInfo.Proxy.Scheme, SingleBrowserInfo.Proxy.Ip, SingleBrowserInfo.Proxy.Port.ToString())
                    };
                    if (!br.GetBrowser().GetHost().RequestContext.SetPreference("proxy", v, out string error))
                        throw new Exception(error);
                };
`

CustomRequestHandler is defined as following:

`public class CustomRequestHandler : RequestHandler
    {
        string _proxyLogin;
        string _proxyPassword;
        string _userAgent;
        public CustomRequestHandler(string proxyLogin, string proxyPassword,string userAgent)
        {
            _proxyLogin = proxyLogin;
            _proxyPassword = proxyPassword;
            _userAgent = userAgent;
        }
        protected override IResourceRequestHandler GetResourceRequestHandler(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, IRequest request, bool isNavigation, bool isDownload, string requestInitiator, ref bool disableDefaultHandling)
        {
            return new CustomResourceRequestHandler(_userAgent);
        }
       
        protected override bool GetAuthCredentials(IWebBrowser browserControl, IBrowser browser, string originUrl, bool isProxy, string host, int port, string realm, string scheme, IAuthCallback callback)
        {
            if (!String.IsNullOrEmpty(_proxyLogin))
            {
                callback.Continue(_proxyLogin, _proxyPassword);
                return true;
            }
            return false;
        }
      
    }`

Now I get load timeout loading with non authorized proxy, while I have successful load using proxy with authorization. What I miss?

My OS is Windows7 64x , VS 2019 installed. I use Offscreen Cefsharp version, .NET 4.5.2 . Cef version is 90.6.50.0

**Comment by amaitland** [comment]
Have you confirmed the proxy works in Chrome? Check the log file to see if there were any errors.

I believe the test cases are currently passing for proxy without auth 

- https://github.com/cefsharp/CefSharp/blob/cefsharp/91/CefSharp.Test/OffScreen/OffScreenBrowserBasicFacts.cs#L562
- https://github.com/cefsharp/CefSharp/blob/cefsharp/91/CefSharp.Test/OffScreen/OffScreenBrowserBasicFacts.cs#L650

**Comment by assopri** [comment]
Thank you! It works just fine!
